packages/rocketchat-language-plugin/server/server.js

```diff
@@ -0,0 +1,82 @@
+const parser = Npm.require('accept-language-parser');
+const languages = Npm.require('languages');
+
+function get_language(languages_codes) {
+
+	const languages_parsed = parser.parse(languages_codes);
+	if (languages_parsed.length===0) {
+		return;
+	}
+	const priority_language = languages_parsed[0].code;
+
+	const priority_language_fullname = languages.getLanguageInfo(priority_language);
+	return priority_language_fullname;
+}
+
+function remove_user_from_automatic_channel() {
+	const user_info = RocketChat.models.Users.findOneById(Meteor.userId());
+	// get the name of channels to which user is added
+	const to_be_removed = user_info.automatic_channels.filter(function(obj) {
+		return obj.blacklisted ==='false';
+	});
+	if (to_be_removed.length!==0) {
```

Comment:
I think this function in general is using too much from global context. You should pass something as arguments. Maybe you should pass which room you are adding the user to and then all other rooms user should be removed from.

You can also try to use `findAndModify` to make a query. Search over Rocket.Chat codebase to see how to use `findAndModify`.